### PR TITLE
break o-date

### LINKED ISSUE #104: o-date complete breaking checklist
Origami plan to release major new versions of low level components, which remove technical debt from recent additions such as component branding, and implement more recent proposals such as adding a primary Sass mixin and removing custom classes from many Sass interfaces.

**steps**

- [x] Add MIGRATION.md
- [x] Add a [primary mixin](https://github.com/Financial-Times/origami-proposals/issues/6), removing other mixins where possible.
- [x] If a mixin is required which does not output a selector, e.g. for component composition, create an [oComponentContent](https://github.com/Financial-Times/origami-proposals/issues/14#issuecomment-485815788) mixin with options like the primary mixin (`Content` infers the mixin does not output selectors).
- [x] Replace methods to customise a component's style globally with a mixin [oComponentCustomise](https://github.com/Financial-Times/origami-proposals/issues/14#issuecomment-485815788) e.g. so a whitelabel user can customise the style of a primary o-buttons button.
- [x] Replace mixins which add new variations of a component with an [oComponentAddSomething](https://github.com/Financial-Times/origami-proposals/issues/14#issuecomment-485815788) mixin e.g. so a user may output a new o-buttons button, alongside primary buttons etc.
- [x] Remove [custom classnames](https://github.com/Financial-Times/origami-proposals/issues/4).
- [x] Use the `$system-code` variable for the `source` param in image service requests, error if not found, update obt to `^ 8.2.2` locally and in circleci config.
- [x] Remove existing semver ranges e.g. for o-brand and o-icons `"o-brand": ">=2.2.0 <4",`. We can't add new features to components used with a semver range without back-porting.
- [x] Use esm consistently over cjs module syntax.
- [x] Solve existing issues labeled "breaking" if possible.

### PR DESCRIPTION
**Pull request by chee**
No sass, so it was moving to esm and adding the migration guide.

closes #104

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,11 +26,17 @@ document.addEventListener('DOMContentLoaded', function() {
 });
 ```
 
-You can also run `require('o-date').init()` once the DOM has loaded if you don't want to initialise other modules at the same time.
+You can also import `o-date` as a component.
+
+```
```
Comment (notlee) [inline README.md]:
```suggestion
```js
```

**Comment by origamiserviceuser** [comment]
o-date bundle size difference from 3.1.5 to 2bedff36208e959c349e9fc4f52b5afe6d74174e
js: 0.39kb increase (0.09kb/gzip)

**Comment by origamiserviceuser** [comment]
o-date bundle size difference from 3.1.5 to 6b9d89ca8771bdc494554f8fb07c14e8b29b06d7
js: 0.39kb increase (0.09kb/gzip)

**Comment by origamiserviceuser** [comment]
o-date bundle size difference from 3.1.5 to 6b20ed68cd16fc43c7d9d65c4f1401a2ef85f4d9
js: 0.39kb increase (0.09kb/gzip)

**Pull request merged**